jupyter/sum-to-zero/sum_to_zero_evaluation.qmd

```diff
@@ -0,0 +1,1582 @@
+i.e., estimates regression coefficients given the observed data.
+We use the generated data as the inputs.
+Because the output files are real-valued outputs, regardless of variable element type,
+model data variables of type int need to be cast to int.
+Here all the observed data is count and categorial data.
+
+```{python}
+data_fixed = {'N':sim_data.pos_tests.shape[1], 
+              'N_age':gen_data_dict['N_age'], 
+              'N_eth':gen_data_dict['N_eth'],
+              'N_edu':gen_data_dict['N_edu'],
+              'sens': gen_data_dict['sens'],
+              'spec': gen_data_dict['spec'],
+              'intercept_prior_mean': gen_data_dict['baseline'],
+              'intercept_prior_scale': 2.5}
+
+data_small = data_fixed | {'pos_tests':sim_data.pos_tests[0].astype(int),
+             'tests':sim_data.tests[0].astype(int),
+             'sex':sim_data.sex[0].astype(int),
+             'age':sim_data.age[0].astype(int), 
+             'eth':sim_data.eth[0].astype(int),
+             'edu':sim_data.edu[0].astype(int)}
+
+data_large = data_fixed | {'pos_tests':sim_data_lg.pos_tests[0].astype(int),
+             'tests':sim_data_lg.tests[0].astype(int),
+             'sex':sim_data_lg.sex[0].astype(int),
+             'age':sim_data_lg.age[0].astype(int), 
+             'eth':sim_data_lg.eth[0].astype(int),
+             'edu':sim_data_lg.edu[0].astype(int)}
+
+data_tiny = data_fixed | {'pos_tests':sim_data_tiny.pos_tests[0].astype(int),
+             'tests':sim_data_tiny.tests[0].astype(int),
+             'sex':sim_data_tiny.sex[0].astype(int),
+             'age':sim_data_tiny.age[0].astype(int), 
+             'eth':sim_data_tiny.eth[0].astype(int),
+             'edu':sim_data_tiny.edu[0].astype(int)}
+```
+
+Record the data-generating parameters
+
+```{python}
+true_params = {
+    'beta_0': sim_data.beta_0[0],
+    'pct_sex': sim_data.pct_sex[0],
+    'beta_sex': sim_data.beta_sex[0],
+    'pct_age': sim_data.pct_age[0],
+    'beta_age':sim_data.beta_age[0],
+    'pct_eth': sim_data.pct_eth[0],
+    'beta_eth':sim_data.beta_eth[0],
+    'pct_edu': sim_data.pct_edu[0],
+    'beta_edu':sim_data.beta_edu[0]
+}
+true_params
+```
+
+#### Model 1: `sum_to_zero_vector`
+
+This model is in file [binomial_4_preds_ozs.stan](https://github.com/stan-dev/example-models/tree/master/jupyter/sum-to-zero/stan/binomial_4_preds_ozs.stan).
+
+```{python}
+binomial_ozs_mod = CmdStanModel(stan_file=os.path.join('stan', 'binomial_4preds_ozs.stan'))
+```
+```{python}
+binomial_ozs_fit = binomial_ozs_mod.sample(data=data_small, parallel_chains=4)
+```
+Record the seed used for the first run and use it for all subsequent fits.
+```{python}
+a_seed = binomial_ozs_fit.metadata.cmdstan_config['seed']
+```
+
+```{python}
+binomial_ozs_fit_lg = binomial_ozs_mod.sample(data=data_large, parallel_chains=4, seed=a_seed)
+```
+
+#### Model 2:  Hard sum-to-zero constraint
+
```

Comment:
Highlight what has changed from the previous Stan code. I suggest putting all the python into one block. This goes for the following section as well.